### ISSUE TITLE
[1.12.2] Database contents reset on reload with 137 bees stored

### ISSUE THREAD

**Issue by SumaniPardia** [ForestryMC/ForestryMC#2131]
I had filled a database storage (the one made with both an apiarist and arborist chest) with drones while I was breeding several bees.  Upon going to the nether to retrieve something, I found that when I got back my database storage was empty.  I am currently on 5.8.0.270 as part of the Modern Skyblock 3 Departed pack.  I do have a backup from before I went to the nether and the drones are still there.  Please let me know if there is anything else you want me to try.

**Comment by Nedelosk** [comment]
I can't reproduce the bug.

**Comment by Nedelosk** [comment]
Before Nether:
![2018-05-13_14 06 19](https://user-images.githubusercontent.com/10259214/39967155-e89df1cc-56b6-11e8-9152-42ce762a0e52.png)
After Nether:
![2018-05-13_14 06 30](https://user-images.githubusercontent.com/10259214/39967159-edb4ec6a-56b6-11e8-8bca-d276ec3786be.png)

**Comment by SumaniPardia** [comment]
How many slots does the database have?  I can't restore it from my backup as it's empty when I copy it over.  Digging through NBT and I found this:
![image](https://user-images.githubusercontent.com/38802058/39971959-1703ef58-56cb-11e8-9bca-b22332ce875e.png)

**Comment by temp1011** [comment]
Oh fun negative slots. It looks like it's meant to have 136 slots but that was a quick glance.

**Comment by SumaniPardia** [comment]
Hmm, with unsigned integers that would make the -121 slot 137.  Somehow it must have let me store a bee too many, client side lag?

**Comment by slotzero** [comment]
This happened to me as well.  I was using the database for a while and it was only after I managed to fill it up that I noticed all my trees had disappeared.  I didn't enter any dimensions, but about an hour after it was filled I did reboot the MC server to apply updates to the OS.  When I logged back in, the database was empty.  I was using forestry_1.12.2-5.8.1.341 w/ Forge 2749.

**Comment by mconwa01** [comment]
I've encountered this as well, using Forestry 1.12.2-5.8.2.354, Forge 14.23.4.2760. Steps to reproduce in creative mode:

1. Place a database.
2. Put at least 129 separate stacks in it (I used forest queens from the creative inventory).
3. Save/quit and reload the world.
4. Database is empty.

Looks like the database has 136 slots, but when it gets saved to NBT, the slot index for each slot is a (signed) byte, which has a maximum value of 127:

https://github.com/ForestryMC/ForestryMC/blob/7a45b02ca64b3bb952c2723e4f9868f3fcdda80d/src/main/java/forestry/core/utils/InventoryUtil.java#L636

So any slots greater than 127 (i.e. beyond the 128th slot, after 0-indexing), after a NBT round-trip, come back as negative numbers (two's complement, as seen above). That negative index eventually gets passed to NonNullList.set() from here (I think):

https://github.com/ForestryMC/ForestryMC/blob/7a45b02ca64b3bb952c2723e4f9868f3fcdda80d/src/main/java/forestry/core/inventory/InventoryPlain.java#L71

Not sure why that causes the whole inventory to be empty on load; maybe an exception is thrown and silenced somewhere?
